base_exception/models/base_exception.py

```diff
@@ -162,16 +163,16 @@ def detect_exceptions(self):
     def _exception_rule_eval_context(self, obj_name, rec):
         user = self.env['res.users'].browse(self._uid)
         return {obj_name: rec,
-                'self': self.pool.get(rec._name),
```

Comment:
this is not the same thing
```python
>>> self.pool.get('res.users')
<class 'odoo.api.res.users'>
>>> self.env.get('res.users')
res.users()
```
Make sure rules are not impacted. An option can be to keep `self` and deprecated until v12 and use `model` key for the new one.